tests/tr1/tests/valarray/test.cpp

```diff
@@ -340,5 +340,24 @@ void test_main() { // test basic workings of valarray definitions
     CHECK_INT(v0.max(), 4);
     CHECK_INT(v0.min(), 2);
 
+    // test range-based for
+    v0 = Mycont();
+    for (char c : v0) {

```

Comment:
No change requested: I generally recommend that all range-based for-loops should be written with `auto&`, `const auto&`, or (in rare cases where proxies are relevant) `auto&&`. (This avoids the potential harms of copying elements: losing modifications, taking the address of soon-to-be-destroyed objects, etc.) However, there are no actual correctness concerns here, and we don't imitate the `tr1` legacy test suite's style elsewhere, so I don't think it's worth changing this and resetting testing.